### ISSUE TITLE
Throw error in deployed general agent if didn't run to completion

### ISSUE THREAD

**Issue by evangriffiths** [gnosis/prediction-market-agent#372]
Microchain swollows up some errors (e.g. https://github.com/gnosis/prediction-market-agent/issues/354), so it's hard to know if the agent is working correctly. We can add our own check - if the agent hasn't run for `n_iterations` and the last call isn't `Stop`, then we can throw our own exception.

**Comment by kongzii** [comment]
This bugged me in CrewAI so much as well. Would it make sense to you to add PR into Microchain, that would optionally throw the exceptions instead of silencing them into a string output?

**Comment by evangriffiths** [comment]
Yes I think that's a better idea than this! So adding `throw_on_abort` arg to `Agent.step` [here](https://github.com/galatolofederico/microchain/blob/main/microchain/engine/agent.py#L70).

I think it would need a little refactoring:
- abort currently happens if (1) llm response is empty, or no function call is found in the reply (after max_retries), (2) the function call throws an exception (after max_retries), or (3) the agent calls `Stop()`. We only want to throw for (1) and (2)
- need to make sure `reply` has the desired message at the time of breaking inside `Agent.step`. I think this will be okay after https://github.com/galatolofederico/microchain/issues/14 actually.
- for abort type (1) there is no `output` returned from `step`, but for (2) there is. So these cases would need to be handled differently

**Comment by evangriffiths** [comment]
Fixed by https://github.com/gnosis/prediction-market-agent/pull/396
